raft/server.cc

```diff
@@ -687,22 +687,19 @@ future<add_entry_reply> server_impl::execute_modify_config(server_id from,
         // `modify_config` doesn't actually need the entry id for anything
         // but we reuse the `add_entry` RPC verb which requires it.
         co_return add_entry_reply{entry_id{}};
+    } catch (not_a_leader& e) {
+        co_return add_entry_reply{transient_error{std::current_exception(), e.leader}};
+    } catch (dropped_entry&) {
+        co_return add_entry_reply{transient_error{std::current_exception(), {}}};
+    } catch (conf_change_in_progress&) {
+        co_return add_entry_reply{transient_error{std::current_exception(), {}}};
```

Comment:
I find this change asking for trouble. There may be a common error handling path, like the code checking for is_uncertainty() can get an else {} branch. I find the motivation for the patch provided in the pull request weak. However, I am not willing to fight for it, if you feel strongly about this change I don't want to bikeshed.

Reply:
i don't feel strong. just thought that we should use RTTI only if necessary. actually, i thought about trading `is_uncertainty()` for a common base class of the uncertain-ed exceptions. but didn't want to go that far as i am not familiar with those exceptions and how they are supposed to be used.

Reply:
anyway, i see you point. if any of us also has similar feelings, let's revisit this option then.